nightfall-optimist/src/services/transaction-sign-send.mjs

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-param-reassign */
+
+import config from 'config';
+import { web3 } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import { estimateGas, estimateGasPrice } from '@polygon-nightfall/common-files/utils/gas.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { Mutex } from 'async-mutex';
+
+const { GAS, GAS_MULTIPLIER, GAS_ESTIMATE_ENDPOINT, GAS_PRICE, GAS_PRICE_MULTIPLIER } = config;
+
+const nonceMutex = new Mutex();
+
+const isWeb3Listening = async () => {
+  try {
+    const isListening = await web3.eth.net.isListening();
```

Comment:
You can return directly.

Reply:
Like we said it is not actually possible to return straight from an `asyn` call within a `try/catch` block (see https://jakearchibald.com/2017/await-vs-return-vs-return-await/).

In any case - like you pointed out - both methods `isWeb3Listening` and `getAddressNonce` were used locally and I've ended up removing both.

Thanks

Reply:
Amazing, thank you!